pandas/io/common.py

```diff
@@ -583,6 +587,9 @@ def write(self, data):
         archive_name = self.filename
         if self.archive_name is not None:
             archive_name = self.archive_name
+        if archive_name is None:
+            # ZipFile needs to a non-empty string
+            archive_name = "zip"
```

Comment:
Needed for `df.to_csv(io.BytesIO(), compression="zip", mode="wb")`. This was not covered in [this test](https://github.com/pandas-dev/pandas/blob/59ffa251269ddb8b561d8d88f631a5db21f660dc/pandas/tests/io/test_compression.py#L117).